packages/lab/stories/file-drop-zone/file-drop-zone.stories.tsx

```diff
@@ -1,336 +1,172 @@
 
 import { Meta, StoryFn } from "@storybook/react";
-import { useDensity } from "@salt-ds/core";
 import {
-  createFileTypeValidator,
-  createTotalSizeValidator,
+  Banner,
+  BannerContent,
+  Button,
+  StackLayout,
+  Text,
+  Tooltip,
+} from "@salt-ds/core";
+import {
   FileDropZone,
-  FilesAcceptedEventHandler,
-  FilesRejectedEventHandler,
-  FilesValidator,
+  FileDropZoneIcon,
+  FileDropZoneIconProps,
+  FileDropZoneProps,
+  FileDropZoneTrigger,
+  FileDropZoneTriggerProps,
 } from "@salt-ds/lab";
 import { AllRenderer } from "docs/components";
+import {
+  containsFiles,
+  createFileTypeValidator,
+  createTotalSizeValidator,
+  extractFiles,
+  FilesValidator,
+  validateFiles,
+} from "./utils";
 
 export default {
   title: "Lab/File Drop Zone",
   component: FileDropZone,
 } as Meta<typeof FileDropZone>;
 
-export const All: StoryFn<typeof FileDropZone> = ({
-  onFilesAccepted,
-  ...args
-}) => {
+export const All: StoryFn<typeof FileDropZone> = ({ onDrop, ...args }) => {
   return (
     <AllRenderer>
-      <FileDropZone {...args} />
-    </AllRenderer>
-  );
-};
-
-export const Default: StoryFn<typeof FileDropZone> = ({
-  onFilesAccepted,
-  ...args
-}) => {
-  const [result, setResult] = useState<{ files: readonly File[] }>();
-
-  const handleFilesAccepted = useCallback<FilesAcceptedEventHandler>(
-    (files, event) => {
-      console.log("onFilesAccepted:", { files, event });
-      setResult({ files });
-
-      onFilesAccepted?.(files, event);
-    },
-    [onFilesAccepted]
-  );
-
-  return (
-    <>
-      <FileDropZone
-        data-testid="file-drop-zone-example"
-        onFilesAccepted={handleFilesAccepted}
-        {...args}
-      />
-      <ResultCard result={result} />
-    </>
-  );
-};
-Default.args = {};
-
-export const Disabled: StoryFn<typeof FileDropZone> = ({
-  onFilesAccepted,
-  ...args
-}) => {
-  const handleFilesAccepted = useCallback<FilesAcceptedEventHandler>(
-    (files, event) => {
-      console.log("onFilesAccepted:", { files, event });
-
-      onFilesAccepted?.(files, event);
-    },
-    [onFilesAccepted]
-  );
-
-  return (
-    <FileDropZone
-      data-testid="file-drop-zone-example"
-      disabled
-      onFilesAccepted={handleFilesAccepted}
-      {...args}
-    />
-  );
-};
-Disabled.args = {};
-
-export const WithCustomText: StoryFn<typeof FileDropZone> = ({
-  onFilesAccepted,
-  ...args
-}) => {
-  const [result, setResult] = useState<{ files: readonly File[] }>();
-
-  const handleFilesAccepted = useCallback<FilesAcceptedEventHandler>(
-    (files, event) => {
-      console.log("onFilesAccepted:", { files, event });
-      setResult({ files });
-
-      onFilesAccepted?.(files, event);
-    },
-    [onFilesAccepted]
-  );
-
-  return (
-    <>
-      <FileDropZone
-        data-testid="file-drop-zone-example"
-        onFilesAccepted={handleFilesAccepted}
-        {...args}
-      >
-        Some custom text
+      <FileDropZone {...args}>
+        <FileDropZoneIcon />
```

Comment:
What's the logic behind passing this down to the FileDropZone and Trigger as children instead of heaving this within the File Dropzone Component and passing down props ?

Reply:
Salt's ethos is composable APIs